enhancements/network/networking-console-plugin.md

```diff
@@ -0,0 +1,149 @@
+---
+title: networking-console-plugin
+authors:
+  - "@upalatucci"
+reviewers:
+  - "@orenc1"
+  - "@zshi-redhat"
+  - "@pcbailey"
+  - "@metalice"
+  - "@spadgett"
+  - "@jhadvig"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2024-04-12
+last-updated: 2024-04-12
+tracking-link:
+  - https://issues.redhat.com/browse/CONSOLE-3952
+---
+
+
+# Create console plugin for the networking section
+
+
+
+## Summary
+
+
+At present, several pages within the networking section of our console are defined within the console repository. 
+However, working with the console repository has slowed down development and there are multiple motivations to convert static console plugins into dynamic ones described [here](https://github.com/spadgett/enhancements/blob/master/enhancements/console/dynamic-plugins.md#motivation). 
+To address this, we propose relocating the relevant code to a separate repository named "networking-console-plugin" and integrating it into the cluster-networking-operator.
+Specifically: Services, Route, Ingress, NetworkPolicy, and NetworkAttachmentDefinition pages for listing, creating, and editing those resources.
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### User Stories
+
+N/A. For the end user the UI will be the same. The cluster admin will have the option to disable and enable the plugin but it's not relevant. 
+
+
+### Goals
+
+
+Enhanced Development Efficiency: By isolating networking console pages into a dedicated repository, developers can expedite development cycles and foster faster iterations, ultimately enhancing overall development efficiency.
+
+Improved Maintainability: Segregating the networking console pages from the main console repository will simplify maintenance and testing efforts.
+
+Dedicated Team: Currently, backend networking features are progressing faster than the corresponding UI console pages. This change will make the CNV UI team directly responsible for enhancing pages and making future features easy to use and discover.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+To execute the migration of networking console pages to the "networking-console-plugin" repository and integrate them into the cluster-networking-operator, we propose the following steps:
+
+Create "networking-console-plugin" Repository (Done): Establish a new repository named "networking-console-plugin" to house the codebase for networking console pages. This repository will serve as the centralized location for all networking-related functionalities.
+
+Move Networking Console Pages (Done): Transfer relevant code files and resources from the console repository to the "networking-console-plugin" repository and refactor them. Ensure accurate migration of dependencies and configurations to maintain seamless functionality.
```

Comment:
How does this change impact the topology view where we have a details sidebar for networking resources like services and routes?

@jerolimov FYI

Reply:
At the moment we don't impact this part. We just re-define the list, details and creation pages

Reply:
we have a [pr](https://github.com/openshift/console/pull/13675) in place to remove those pages from the console code.